src/compiler/program.ts

```diff
@@ -1076,11 +1076,6 @@ namespace ts {
                 !options.experimentalDecorators) {
                 programDiagnostics.add(createCompilerDiagnostic(Diagnostics.Option_0_cannot_be_specified_without_specifying_option_1, "emitDecoratorMetadata", "experimentalDecorators"));
             }
-
-            if (options.experimentalAsyncFunctions &&
-                options.target !== ScriptTarget.ES6) {

```

Comment:
there should be another message `Async functions are only available when targeting ECMAScript 6 and higher.` which i have not changed